File: Innovation-Challenge/史大伟-AI招聘助手/app/src/main/java/com/kangaroo/ktdemo/data/service/ApiService.kt
package com.kangaroo.ktdemo.data.service

import com.kangaroo.ktdemo.app.AI_GROUP_ID
import com.kangaroo.ktdemo.app.AI_SCRICT
import com.kangaroo.ktdemo.data.model.network.requests.AIRequest
import com.kangaroo.ktdemo.data.model.network.responses.AIResponse
import retrofit2.Response
import retrofit2.http.*

interface ApiService {
//    headers = {"Authorization": f"Bearer {api_key}", "Content-Type": "application/json"}
    @POST(ApiMethods.chatcompletionPro)
    @Headers(
        "Authorization:Bearer $AI_SCRICT",
        "Content-Type:application/json"
    )
    suspend fun chatcompletionPro(@Query("GroupId") groupId:Long = AI_GROUP_ID,@Body payload: AIRequest): AIResponse

}

File: Innovation-Challenge/张秋伟-RickAndMorty角色扮演/lib/data/services/api_service.dart
import 'dart:collection';

import 'package:retrofit/retrofit.dart';
import 'package:rick_and_morty/data/model/params/ai_param.dart';
import 'package:rick_and_morty/data/model/responses/ai_response.dart';
import '../../app/config.dart';
import 'api_methods.dart';
import 'package:dio/dio.dart' hide Headers;
part 'api_service.g.dart';

@RestApi()
abstract class RestClient {
  factory RestClient(Dio dio, {String baseUrl}) = _RestClient;

  @POST(ApiMethods.chatcompletionPro)
  // @Headers(
  //     "Authorization:Bearer $AI_SCRICT",
  //     "Content-Type:application/json"
  // )
  @Headers(<String, dynamic>{
    'Authorization': "Bearer ${VmAppConfig.aiScrict}",
    'Content-Type': 'application/json',
  })
  Future<AiResponse> chatcompletionPro(@Body() AiParma payload,{@Query("GroupId") int groupId = VmAppConfig.aiGroupId});

}
